lib/src/model.dart

```diff
@@ -654,8 +661,11 @@ class Class extends ModelElement
 
   @override
   String get href {
-    if (canonicalLibrary == null) return null;
-    return '${canonicalLibrary.dirName}/$fileName';
+    if (!identical(canonicalModelElement, this))
```

Comment:
nit: braces for the if statement body

Reply:
Just for clarity, if this element is not the canonical one, we generate a link to the canonical one?

Reply:
Correct.  While you could infer that from the code previously with all the "canonicalEnclosingElement" stuff, the modification here is intended to make that more clear and to make the object being linked to actually responsible for its own link generation.

Most of the broken link bugs I've seen either were attributable to bugs in how this worked, or were made harder to fix because of it (like this one).  Now elements no longer point blindly into the filesystem hoping that the model really has a canonical element that matches, but we actually look up the element.

Reply:
I've got a second change on deck to eliminate the copypasta of hrefs everywhere; if possible I'd prefer to fix the braces there.

Reply:
sgtm